osu.Game/Screens/Select/PlaySongSelect.cs

```diff
@@ -174,9 +188,43 @@ private void start()
             Task.Factory.StartNew(() => addBeatmapSets(game, initialAddSetsTask.Token), initialAddSetsTask.Token);
         }
 
+        private ScheduledDelegate filterTask;
+
+        private void filterChanged()
+        {
+            filterTask?.Cancel();
+            filterTask = Scheduler.AddDelayed(() =>
+            {
+                filterTask = null;
+                var search = filter.Search;
+                BeatmapGroup newSelection = null;
+                foreach (var beatmapGroup in carousel)
+                {
+                    var set = beatmapGroup.BeatmapSet;
+                    bool match = string.IsNullOrEmpty(search)
+                        || (set.Metadata.Artist ?? "").IndexOf(search, StringComparison.InvariantCultureIgnoreCase) != -1
+                        || (set.Metadata.ArtistUnicode ?? "").IndexOf(search, StringComparison.InvariantCultureIgnoreCase) != -1
+                        || (set.Metadata.Title ?? "").IndexOf(search, StringComparison.InvariantCultureIgnoreCase) != -1
+                        || (set.Metadata.TitleUnicode ?? "").IndexOf(search, StringComparison.InvariantCultureIgnoreCase) != -1;
+                    if (match)

```

Comment:
The filter will also have to work with individual difficulties; not just groups; in the future. Probably out of score for this PR, too, given that we still need to finalize how sorting e.g. by difficulty will work in the first place.